File: JavaScript/8js 2/mango.js
/*
|--------------------------------------------------------------------------
| Price of Mangoes
|--------------------------------------------------------------------------
|
*/

function mango(quantity, price) {
   return Math.ceil(quantity - quantity / 3) * price;
}


function mango(quantity, price) {
   return (quantity - ~~(quantity / 3)) * price;
}


function mango(quantity, price) {
   return price * (quantity - Math.floor(quantity / 3));
}


function mango(quantity, price) {
   return (quantity - ((quantity / 3) ^ 0)) * price;
}


console.log(mango(2, 3), 6);
console.log(mango(3, 3), 6);
console.log(mango(5, 3), 12);
console.log(mango(9, 5), 30);

File: Python/8py 2/mango.py
"""
|--------------------------------------------------------------------------
| Price of Mangoes
|--------------------------------------------------------------------------
|
"""

def mango(quantity, price):
   return price * (quantity - quantity // 3)


mango = lambda q, p: p * (q - q // 3)


def mango(quantity, price):
   a, b = divmod(quantity, 3)
   return price * (a * 2 + b)


print(mango(2, 3), 6)
print(mango(3, 3), 6)
print(mango(5, 3), 12)
print(mango(9, 5), 30)